### PR TITLE
struct: fix incompatibility error with older versions

### PR DESCRIPTION
**Pull request by YangKian**
### What problem should be fixed?
This fixes issue #218 

### Have you added test cases to catch the problem?
Yes，I add a new test case to cover the situation issue #218 mentioned

### CODE REVIEW

File: struct.go [hunk 1]
```diff
@@ -183,6 +183,10 @@ func setWithProperType(t reflect.Type, key *Key, field reflect.Value, delim stri
 		if vt.Name() == "Duration" {
 			durationVal, err := key.Duration()
 			if err != nil {
+				if intVal, err := key.Int64(); err == nil {
+					field.SetInt(intVal)
+					break
```
Comment (unknwon) [inline struct.go]: I think this is more appropriate:
```suggestion
					return nil
```

Comment (YangKian) [reply struct.go]: Yes, its a better way to use `return nil`, I have fix it, thanks a lot :) So sorry for submit too many commits, I'm green hands for git, tried to merge and modify the commits, but failed.

Comment (unknwon) [reply struct.go]: @YangKian No worries! GitHub has squash merge so it doesn't matter :)

**Comment by unknwon** [comment]
Thank you!

**Pull request merged**